File: backend/src/main/java/com/mycompany/backend/service/QuestionsService.java
package com.mycompany.backend.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.mycompany.backend.dao.QuestionsDao;
import com.mycompany.backend.dto.Pager;
import com.mycompany.backend.dto.Question;

@Service
public class QuestionsService {
   @Autowired
   private QuestionsDao questionsDao;
   
   //데이터 개수
   public int getCount() {
      return questionsDao.count();
   }

   //페이지 단위 리스트 가져오기
   public List<Question> getList(Pager pager) {
      return questionsDao.selectByPage(pager);
   }
   
   //질문번호에 따른 목록
   public Question getQuestion(int question_id) {
      return questionsDao.selectById(question_id);
   }
   
   //질문 수정
   public int update(Question question) {
	  return questionsDao.update(question);
   }
   
   //삭제
   public int delete(int question_id) {
      return questionsDao.deleteById(question_id);
   }
   
}

File: frontend/services/questionsService.js
angular.module("app")
  .factory("questionsService", function($http) {
    //변수 선언
    const BASE_URL = "http://localhost:8080/question";
    //서비스 객체 리턴
    return {
      list: function(pageNo=1) {
        const promise = $http.get(BASE_URL, {params:{pageNo}});
        return promise;
      },
      read: function(question_id) {
        const promise = $http.get(BASE_URL + "/" + question_id);
        return promise;
      },
      update: function(question){
        const promise=$http.put(BASE_URL,question);
        return promise;
      },
      delete: function(question_id){
        const promise=$http.delete(BASE_URL + "/" + question_id);
        return promise;
      }
    }
  });